branchpro/posterior.py

```diff
@@ -248,6 +251,158 @@ def get_intervals(self, central_prob):
         return intervals_df
 
 
+#
+# BranchProPosteriorMultSI Class
+#
+
+
+class BranchProPosteriorMultSI(BranchProPosterior):
+    r"""BranchProPosteriorMultiSI Class:
+    Class for computing the posterior distribution used for the inference of
+    the reproduction numbers of an epidemic in the case of a branching process
+    using mutiple serial intevals. Based on the :class:`BranchProPosterior`.
+
+    Parameters
+    ----------
+    inc_data
+        (pandas Dataframe) contains numbers of new cases by time unit (usually
+        days).
+        Data stored in columns of with one for time and one for incidence
+        number, respectively.
+    daily_serial_intervals
+        (list of lists) List of unnormalised probability distributions of that
+        the recipient first displays symptoms s days after the infector first
+        displays symptoms.
+    alpha
+        the shape parameter of the Gamma distribution of the prior.
+    beta
+        the rate parameter of the Gamma distribution of the prior.
+    time_key
+        label key given to the temporal data in the inc_data dataframe.
+    inc_key
+        label key given to the incidental data in the inc_data dataframe.
+    """
+    def __init__(
+            self, inc_data, daily_serial_intervals, alpha, beta,
+            time_key='Time', inc_key='Incidence Number'):
+
+        super().__init__(
+            inc_data, daily_serial_intervals[0], alpha, beta, time_key,
+            inc_key)
+
+        for si in daily_serial_intervals:
+            try:
```

Comment:
This code is now repeated three times. What if we added a method to the base class
```
def _check_serial(self, si):
    try:
        float(next(iter(si)))
    except (TypeError, StopIteration):
        raise TypeError(
            'Daily Serial Interval distributions must be iterable')
    except ValueError:
        raise TypeError('Daily Serial Interval distribution must contain \
                        numeric values')
```
and the run the checks using
```
        for si in daily_serial_intervals:
            self._check_serial(si)
```
This would require that you change back the exception message in the tests.